Yosemite/Yosemite/Model/Storage/Product+ReadOnlyConvertible.swift

```diff
@@ -125,8 +125,8 @@ extension Storage.Product: ReadOnlyConvertible {
                        name: name,
                        slug: slug,
                        permalink: permalink,
-                       date: date ?? Date(),
-                       dateCreated: dateCreated ?? Date(),
+                       date: date ?? Date(timeIntervalSince1970: 0),
+                       dateCreated: dateCreated ?? Date(timeIntervalSince1970: 0),
```

Comment:
> The default date to be applied is 00:00:00 UTC on 1 January 1970 instead of the current date. This is also applied to Product and ProductImage for testability and consistency.

Could you please explain why we need to avoid using the current date as the default date?

Reply:
This is for testability mostly as it's impossible to get a matching Date for the expectations. This is a fallback value so it doesn't matter what the value is IMO.